src/views/connect_dialog_handler.rs

```diff
@@ -509,8 +514,15 @@ impl ConnectDialog {
                             .style(CONNECTION_ERROR_DISPLAY.get_text_color()),
                         text("Node Id").size(12),
                         {
-                            let mut node_input =
-                                text_input("Enter node id", &self.nodeid).padding(5);
+                            let mut node_input = text_input("Enter node id", &self.nodeid)
+                                .padding(5)
+                                .id(IROH_INPUT_ID.clone())
```

Comment:
Could you get the id after creating it, and store it in a Option<Id> field of the class?
Not sure that is much better, more a question.

Reply:
Just saw some of the examples  having this style of implementation.
So, just thought of following that.